text/3724-theupdateframework.md

```diff
@@ -0,0 +1,341 @@
+
+###### Snapshots & Timestamps Role
+
+These roles shall be a single-member role with a key utilized for automation.
+
+### `rust-lang/tuf-crates`
+
+The actual target for tuf-crates shall be the crates index and not the artifacts themselves. This means that the TUF repository for crates.io is performed on much smaller payloads, which still provides us with cryptographic security due to the fact the index contains SHA-256 hashes of the crate file artifacts.  Given the index already consists of SHA-512 signatures of all files, we are then utilizing TUF to validate the index, which in turn is utilized to validate the actual downloaded artifacts. This allows us to perform validation on index updates and not on final downloads, also reducing the overhead of performing multiple hashing and validation procedures on the larger crate artifact files.
+
+#### Roles
+
+##### Root Role
+
+The root role of the tuf-crates repository shall consist of all members of the crates.io rust team with a threshold of 3. As a special case, updating this role shall also require a resigning by the root role of the tuf-root repository (sign a metadata entry existing within tuf-root). This means any changes to the membership of the crates.io team will also require a signing ceremony via github by the root quorum.
+
+##### Targets Role
+
+The repository shall have a top-level Targets role which is utilized directly for all actions on crates.io. This role shall be a 1-key member role, allowing for the automation of actions by crates.io for publishing of crates. In the future, we propose the delegation of roles for trusted publishing within organizations or crates.
+
+###### Snapshots & Timestamps Roles
+
+These roles shall each be a single-member role with a key utilized for automation.
+
+## TUF Management
+
+We propose the adaptation and implementation of TUF-on-CI (https://github.com/theupdateframework/tuf-on-ci) to manage roots and signing events via GitHub CI. This provides a GitHub-centric workflow for performing signing ceremonies via Pull Requests directly on the TUF repositories in question.
+
+Online signing needs shall be implemented with AWS KMS.
+
+All members of all signing quorums within the Rust Project will require hardware keys, the expenses for which will be covered by the Rust Foundation.
+
+## Root Quorum Model
+
+The root key shall follow a `5-of-9` authentication model for all operations. We consider this a reasonable middle ground of quorum to prevent malicious activity, while allowing for rapid response to an event requiring a quorum. These events are iterated below in [When the Quorum will be needed][when-the-quorum-will-be-needed].
```

Comment:
just to play devil's advocate here: what would happen the majority of keyholders met at a large event and were all suddenly incapacitated? is there a way to recover from such a worst-case scenario?